### DISCUSSION TITLE
SD card burning through USB

### DISCUSSION THREAD

**Discussion by shodan8192** [gtxzsxxk/duo-buildroot#15] [Q&A] [answered]
As the `duo-datasheet-v1.2` state :
> USB C port for power and data (and for reprogramming the TF CARD)

Do you know a method for writing new image to SD card, without pulling it from Milk-V Duo ?

**Comment by gtxzsxxk** [comment]
Hello, I am sorry that I am not getting your point. Could you please tell me that what does the pulling from milkv duo mean? Thanks!

**Reply by shodan8192** [reply]
By pulling from milkv duo I mean removing SD card to burn it in external reader. It is possible to write new image to card, while it's still in board socket ?

The idea is to let milkv duo act as usb mass storage device allowing access to SD card, of course not when linux is running, but from uboot maybe.

**Reply by gtxzsxxk** [reply]
Technically that is possible. u-boot should have a good support for the usb. Maybe you can try the gadget to make the sdcard be your mass storage device. But note that if you reprogram the sd card, the previous firmware to be loaded in the IRAM will be gone. In all, this is crazy, unstable and dangerous. But it is worth playing😂.

**Reply by shodan8192** [reply, marked as answer]
OK, let's leave it for the future then 😀
